### ISSUE TITLE
One missing .d file on 32-bit

### ISSUE THREAD

**Issue by andreasbaumann** [ccache/ccache#248]
Running the tests on Archlinux32 I get one failing test:

```
Failure reason: Found 158 (expected 159) *.d files in /root/ccache/src/ccache-3.4.2/testdir.2285/.ccache
```

find testdir.2285/ -type f -name '*.d' | wc -l
=> 158 files

I attached the whole testdir:

[testdir-2285.zip](https://github.com/ccache/ccache/files/1887140/testdir-2285.zip)

BTW: In order to further debug the issue, how can I keep the testdir on 64-bit in order to do a diff
between the two tests?

**Comment by afbjorklund** [comment]
If people are still using 32-bit in 2018, maybe we should add an `-m32` to the build matrix...

**Comment by andreasbaumann** [comment]
It could be RISC-V 32-bit, ARM 32-bit, etc. just because IA-32 is on it's way out, doesn't mean we
should only program for an 64-bit/Linux/Intel microcosmos (or rather macrocosmos). :-)

**Comment by afbjorklund** [comment]
I don't think it will be a big deal to support (most compiler toolchains still have multilib out-of-the-box)
And as long as we are supporting legacy DOS stuff like MinGW32, then 32-bit Linux isn't weird at all...

Will add it to Travis, see if it reproduces

**Comment by andreasbaumann** [comment]
Thanks.. and if I find out how I can keep and diff the test results, am more than happy to debug.. :-)

**Comment by afbjorklund** [comment]
Good news is that it was easy to reproduce.

**Comment by afbjorklund** [comment]
Just as suspected, it is some kind of roundoff. Apparently limit 0.9 is _right_ on the edge...

32-bit: `Threshold: 26.999999`
64-bit: `Threshold: 27.000000`

So we should fix the rounding, if we want to make the cleanup results more predictable.

**Comment by andreasbaumann** [comment]
Glad, it's reproducable and easy to fix. :-)

Out of curiousity: why are files counted as floats?

**Comment by andreasbaumann** [comment]
..ah, I see, it's a roundf after a calculation.

**Comment by mich181189** [comment]
I'm hitting this on Fedora as well - the Fedora ccache package is built for 32-bit.
 
Only remaining question is this: why is this ticket referencing commits that aren't on any branch? seems like an awkward way to use Git!

**Comment by afbjorklund** [comment]
Fixed in 390035abfa4e174ac0b1528435e2d001d39654a8 and 359a4078a8adad86bf028cd4d48210379eea0f7b (on `3.4-maint)

**Comment by jrosdahl** [comment]
> Fixed in 390035a and 359a407 (on 3.4-maint)

Right, so closing this issue.
